plugin/trino-phoenix5/src/test/java/io/trino/plugin/phoenix5/TestPhoenixTypeMapping.java

```diff
@@ -717,4 +780,18 @@ private Session sessionWithDecimalMappingStrict(UnsupportedTypeHandling unsuppor
                 .setCatalogSessionProperty("phoenix", UNSUPPORTED_TYPE_HANDLING, unsupportedTypeHandling.name())
                 .build();
     }
+
+    private void assertPhoenixQueryFails(@Language("SQL") String sql, String expectedMessage)
+    {
+        assertPhoenixQueryFails(sql, expectedMessage, true);
+    }
+
+    private void assertPhoenixQueryFails(@Language("SQL") String sql, String expectedMessage, boolean hasCause)
+    {
+        AbstractThrowableAssert throwableAssert = assertThatThrownBy(() -> new PhoenixSqlExecutor(phoenixServer.getJdbcUrl()).execute(sql));
+        if (hasCause) {
+            throwableAssert = throwableAssert.getCause();
+        }
+        throwableAssert.hasMessageContaining(expectedMessage);
+    }
```

Comment:
Let's keep this method simple and change `testUnsupportedBigInt()` instead:
```java
    @Test
    public void testUnsupportedBigInt()
    {
        try (TestTable table = new TestTable(new PhoenixSqlExecutor(phoenixServer.getJdbcUrl()), "tpch.test_unsupported_bigint", "(data bigint, pk bigint primary key)")) {
            assertPhoenixQueryFails(
                    format("INSERT INTO %s VALUES (-9223372036854775809, 1)", table.getName()), // min - 1
                    "ERROR 203 (22005): Type mismatch. DECIMAL and BIGINT for expression: -9223372036854775809 in column 0.DATA");

            // Phoenix JDBC driver throws ArithmeticException instead of SQLException when the value is larger than max of bigint
            assertThatThrownBy(() -> new PhoenixSqlExecutor(phoenixServer.getJdbcUrl()).execute(format("INSERT INTO %s VALUES (9223372036854775808, 2)", table.getName()))) // max + 1
                    .isInstanceOf(ArithmeticException.class)
                    .hasMessage("Overflow");
        }
    }

    private void assertPhoenixQueryFails(@Language("SQL") String sql, String expectedMessage)
    {
        assertThatThrownBy(() -> new PhoenixSqlExecutor(phoenixServer.getJdbcUrl()).execute(sql))
                .getCause()
                .hasMessageContaining(expectedMessage);
    }
```

Reply:
done